### PR TITLE
Start pod infra container when pod is created

### PR DESCRIPTION
**Pull request by baude**
When we create a pod that also has an infra container, we should
start the infra container automatically.  This allows users to add
running containers to the pod immediately.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: libpod/runtime_pod_linux.go [hunk 1]
```diff
@@ -109,6 +109,19 @@ func (r *Runtime) NewPod(ctx context.Context, options ...PodCreateOption) (*Pod,
 		if err := pod.save(); err != nil {
 			return nil, err
 		}
+
+		// Once the pod infra container has been created, we start it
+		if err := ctr.Start(ctx); err != nil {
+			// If the infra container does not start, we need to remove the container
+			// and tear the pod down.
+			if err2 := r.RemoveContainer(ctx, ctr, true); err != nil {
```
Comment (mheon) [inline libpod/runtime_pod_linux.go]: Lowercase removeContainer() so we don't run into a deadlock here

Comment (mheon) [reply libpod/runtime_pod_linux.go]: Actually - I don't think there's a point to doing this separately if we're force-removing the pod below - it should get rid of the containers in the pod, so we don't need an explicit call to RemoveContainer?

Comment (baude) [reply libpod/runtime_pod_linux.go]: Ok, fixed up!

**Comment by mheon** [comment]
LGTM

**Comment by baude** [comment]
bot, retest this please

**Comment by TomSweeneyRedHat** [comment]
LGTM assuming happy tests

**Comment by rhatdan** [comment]
Does this mean the pause container gets launched when I do 

podman pod create

as opposed to having to do 
podman pod start -l

After the pod was created?

**Comment by mheon** [comment]
@rhatdan Yes.

We should probably reference CRI-O to see when they start theirs.

**Comment by mheon** [comment]
It looks like CRI-O starts the infra container at pod creation - like this PR would do. I think we're good to merge here.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Comment by mheon** [comment]
@rh-atomic-bot retry

**Pull request closed without merging**